components/cloudflare_api_key/actions/cloudflare-list-dns-records/cloudflare-list-dns-records.mjs

```diff
@@ -0,0 +1,162 @@
+import cloudflare from "../../cloudflare_api_key.app.mjs";
+import constants from "../../constants.mjs";
+
+const {
+  ANY: DNS_RECORD_MATCH_ANY,
+  ALL: DNS_RECORD_MATCH_ALL,
+} = constants.DNS_RECORD_MATCHES;
+
+const {
+  TYPE: DNS_RECORD_ORDER_TYPE,
+  NAME: DNS_RECORD_ORDER_NAME,
+  CONTENT: DNS_RECORD_ORDER_CONTENT,
+  TLL: DNS_RECORD_ORDER_TLL,
+  PROXIED: DNS_RECORD_ORDER_PROXIED,
+} = constants.DNS_RECORD_ORDER;
+
+const {
+  ASC: DNS_RECORD_DIRECTION_ASC,
+  DESC: DNS_RECORD_DIRECTION_DESC,
+} = constants.DNS_RECORD_DIRECTION;
+
+export default {
+  key: "cloudflare-list-dns-records",
+  name: "List DNS Records",
+  description: "List, search, sort, and filter a zones' DNS records. [See the docs here](https://api.cloudflare.com/#dns-records-for-a-zone-list-dns-records)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    cloudflare,
+    zoneIdentifier: {
+      propDefinition: [
+        cloudflare,
+        "zoneIdentifier",
+      ],
+    },
+    match: {
+      type: "string",
+      label: "Match",
+      description: "Whether to match all search requirements or at least one (any)",
+      options() {
+        return [
+          {
+            label: "Any",
+            value: DNS_RECORD_MATCH_ANY,
+          },
+          {
+            label: "All",
+            value: DNS_RECORD_MATCH_ALL,
+          },
+        ];
+      },
+      optional: true,
+    },
+    name: {
+      propDefinition: [
+        cloudflare,
+        "dnsName",
+      ],
+      optional: true,
+    },
+    order: {
+      type: "string",
+      label: "Order",
+      description: "Field to order records by",
+      options() {
+        return [
+          {
+            label: "Type",
+            value: DNS_RECORD_ORDER_TYPE,
+          },
+          {
+            label: "Name",
+            value: DNS_RECORD_ORDER_NAME,
+          },
+          {
+            label: "Content",
+            value: DNS_RECORD_ORDER_CONTENT,
+          },
+          {
+            label: "TTL",
+            value: DNS_RECORD_ORDER_TLL,
+          },
+          {
+            label: "Proxied",
+            value: DNS_RECORD_ORDER_PROXIED,
+          },
+        ];
+      },
+      optional: true,
+    },
+    page: {
+      type: "integer",
+      label: "Page",
+      description: "Page number of paginated results",
+      optional: true,
+    },
+    perPage: {
+      type: "integer",
+      label: "Per Page",
+      description: "Number of DNS records per page",
+      optional: true,
+    },
```

Comment:
Pipedream components should handle pagination from within:
https://pipedream.com/docs/components/guidelines/#handle-pagination

And since we will be handling pagination internally in this action, some optional props like `order`, `direction`, and maybe a few others can be removed

Reply:
hello @andrewjschuang, missed some of the observations earlier.
Could you confirm that? The pagination handling in situations like this would be to iterate over all pages and return all records, is that right?